src/importBlobAsBulk.js

```diff
@@ -0,0 +1,114 @@
+import {createLogger} from '@natlibfi/melinda-backend-commons';
+import {BLOB_UPDATE_OPERATIONS} from '@natlibfi/melinda-record-import-commons';
+import createDebugLogger from 'debug';
+import {readFromQueue} from './fromQueueHandler';
+
+export default function (mongoOperator, amqpOperator, melindaRestApiClient, config) {
+  const debug = createDebugLogger('@natlibfi/melinda-record-import-importer:importBlobAsBulk');
+  const logger = createLogger(); // eslint-disable-line no-unused-vars
+
+  const {pullState, importOptions} = config;
+  debug(`Import options: ${JSON.stringify(importOptions)}`);
+  return {startHandling};
+
+  async function startHandling(blobId) {
+    try {
+      const {correlationId} = await mongoOperator.readBlob({id: blobId});
+      debug(`Got queue item ${blobId}`);
+      const messageCount = await amqpOperator.countQueue({blobId, status: pullState});
+      debug(`${messageCount} messages in queue ${blobId}.${pullState}`);
+
+      const hasMessages = messageCount > 0;
+      const hasCorrelationId = correlationId !== '';
```

Comment:
What happens id correlationId is for some reason undefined?

Reply:
https://github.com/NatLibFi/melinda-record-import-commons-js/blob/73a0e2a9867db0b019c8b7d4416bd6ab3b8d2366/src/mongoBlobs.js#L134 not possible